### PR TITLE
Feature/complete workflow

### PR DESCRIPTION
**Pull request by wovalle**

### CODE REVIEW

**Review by vetom — APPROVED** [review]

File: src/js/components/Wizard.js [hunk 1]
```diff
@@ -1,22 +1,79 @@
...
+    this.setState(Object.assign({}, this.state, {players}));
+  }
+
+  setCurrentDeck(currentDeck) {
+    const currentCards = werewolfService.getCardsInDeck(currentDeck)
+    .map(c => {
+      return {
+        key: c.key,
+        visible: werewolfService.isInDeck(c.key, currentDeck),
+        value: 1,
```
Comment (wistcc) [inline src/js/components/Wizard.js:32]: I think that this is not correct, isn't it?

File: src/js/pages/Cards.js [hunk 2]
```diff
@@ -2,18 +2,36 @@ import React from 'react';
...
+      .map(c => {
+        return this.props.getCardsInGame().find(k => k.key === c.key) || c;
+      })
       .map(c => (
-        <div class="col-md-6 col-xs-12" key={c.role}>
-          <div class="col-md-8 col-xs-6">
-            <label><input type="checkbox" name={c.role} /> {c.role}</label>
-          </div>
+        <div class="col-md-6 col-xs-12" key={c.key}>
+            <label><input type="checkbox" onChange={this.setVisibility} checked={c.visible || false} name={c.key} /> {c.key}</label>
```
Comment (wistcc) [inline src/js/pages/Cards.js]: `c.visible || false` is not enough `c.visible`?

File: src/js/components/Wizard.js [hunk 3]
```diff
@@ -1,22 +1,79 @@
 import React, {cloneElement} from 'react';
+import { browserHistory } from 'react-router';
 import werewolfService from '../services/werewolf';
 
+const DEFAULT_GAME_MODE = 'balanced';
```
Comment (wistcc) [inline src/js/components/Wizard.js:5]: Probably we should create another file with this `'balanced`', 'chaos` and `'basic'`.

File: src/js/pages/Cards.js [hunk 4]
```diff
@@ -29,9 +47,9 @@ export default class Cards extends React.Component {
...
-              <Link to={{pathname:"game", query:{type: "chaos"} }} className="btn btn-danger pull-right col-md-4 col-xs-12 btn-space"><i class="fa fa-arrows" aria-hidden="true"></i> Start Chaos </Link>
-              <button onClick={browserHistory.goBack} className="btn btn-default pull-left col-md-2 col-xs-12 btn-space"><i class="fa fa-arrow-left" aria-hidden="true"></i> Back</button>
+              <button onClick={browserHistory.goBack} className="btn btn-default col-md-2 col-xs-12 btn-space"><i class="fa fa-arrow-left" aria-hidden="true"></i></button>              
+              <button onClick={this.props.startGame.bind(this, "chaos")} className="btn btn-default col-md-4 col-md-offset-1 col-xs-12 btn-space"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</button>
+              <button onClick={this.props.startGame.bind(this, "balanced")} className="btn btn-default col-md-4 col-md-offset-1 col-xs-12 btn-space"><i class="fa fa-balance-scale" aria-hidden="true"></i> Quick Balanced</button>
```
Comment (wistcc) [inline src/js/pages/Cards.js:52]: Probably we should create another file with this `'balanced`', 'chaos` and `'basic'`.

File: src/js/pages/Home.js [hunk 5]
```diff
@@ -44,12 +44,12 @@ export default class Home extends React.Component {
...
             <div class="col-md-12">
-              <Link to="cards" className="btn btn-success btn-block">Next  <i class="fa fa-arrow-right" aria-hidden="true"></i></Link>
+              <Link to="cards" className="btn btn-success btn-block">Customize  <i class="fa fa-cog" aria-hidden="true"></i></Link>
             </div>
             <hr />
             <div class="col-md-12">
-              <Link to={{pathname:"game", query:{type: "chaos"} }} className="btn btn-default col-md-4 pull-left"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</Link>
-              <Link to={{pathname:"game", query:{type: "balanced"} }} className="btn btn-default col-md-4 pull-right"><i class="fa fa-balance-scale" aria-hidden="true"></i> Quick Balanced</Link>
+              <button onClick={this.props.startGame.bind(this, "chaos")} className="btn btn-default col-md-5"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</button>
+              <button onClick={this.props.startGame.bind(this, "balanced")} className="btn btn-default col-md-5 col-md-offset-2"><i class="fa fa-balance-scale" aria-hidden="true"></i> Quick Balanced</button>
```
Comment (wistcc) [inline src/js/pages/Home.js:52]: Probably we should create another file with this `'balanced`', 'chaos` and `'basic'`.

File: src/js/pages/Cards.js [hunk 6]
```diff
@@ -29,9 +47,9 @@ export default class Cards extends React.Component {
             </div>
             
             <div class="panel-footer">
-              <Link to={{pathname:"game", query:{type: "balanced"} }} className="btn btn-success pull-right col-md-4 col-xs-12 btn-space"><i class="fa fa-balance-scale" aria-hidden="true"></i> Start Balanced </Link>
-              <Link to={{pathname:"game", query:{type: "chaos"} }} className="btn btn-danger pull-right col-md-4 col-xs-12 btn-space"><i class="fa fa-arrows" aria-hidden="true"></i> Start Chaos </Link>
-              <button onClick={browserHistory.goBack} className="btn btn-default pull-left col-md-2 col-xs-12 btn-space"><i class="fa fa-arrow-left" aria-hidden="true"></i> Back</button>
+              <button onClick={browserHistory.goBack} className="btn btn-default col-md-2 col-xs-12 btn-space"><i class="fa fa-arrow-left" aria-hidden="true"></i></button>              
+              <button onClick={this.props.startGame.bind(this, "chaos")} className="btn btn-default col-md-4 col-md-offset-1 col-xs-12 btn-space"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</button>
```
Comment (wistcc) [inline src/js/pages/Cards.js:51]: Probably we should create another file with this `'balanced`', 'chaos` and `'basic'`.

File: src/js/pages/Home.js [hunk 7]
```diff
@@ -44,12 +44,12 @@ export default class Home extends React.Component {
...
           <div class="panel-footer">
             <div class="col-md-12">
-              <Link to="cards" className="btn btn-success btn-block">Next  <i class="fa fa-arrow-right" aria-hidden="true"></i></Link>
+              <Link to="cards" className="btn btn-success btn-block">Customize  <i class="fa fa-cog" aria-hidden="true"></i></Link>
             </div>
             <hr />
             <div class="col-md-12">
-              <Link to={{pathname:"game", query:{type: "chaos"} }} className="btn btn-default col-md-4 pull-left"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</Link>
-              <Link to={{pathname:"game", query:{type: "balanced"} }} className="btn btn-default col-md-4 pull-right"><i class="fa fa-balance-scale" aria-hidden="true"></i> Quick Balanced</Link>
+              <button onClick={this.props.startGame.bind(this, "chaos")} className="btn btn-default col-md-5"><i class="fa fa-arrows" aria-hidden="true"></i> Quick Chaos</button>
```
Comment (wistcc) [inline src/js/pages/Home.js:51]: Probably we should create another file with this `'balanced`', 'chaos` and `'basic'`.

File: src/js/services/werewolf.js [hunk 8]
```diff
@@ -2,10 +2,29 @@ import brain from 'werewolf-brain';
...
 export default new class werewolfService {
     getDecks() {
-        return brain.getAllTemplates();        
+        return brain.getDecks();        
     }
     
     getCards() {
-        return brain.getAllCards();
+        const cards = brain.getCards();
+        return Object.keys(cards).map(_ => {return {key: _, value: cards[_]}});
```
Comment (wistcc) [inline src/js/services/werewolf.js]: I'm a full fan of always having very descriptive names!

**Review by wistcc — CHANGES_REQUESTED** [review]
LVGTM. Just a few comments.

File: src/js/components/Wizard.js (hunk 1 above)
Comment (wovalle) [reply src/js/components/Wizard.js:32]: What do you mean?

**Comment by wovalle** [comment]
@wistcc Since the original motive of this pr was to implement the full workflow of the project, I created two issues with enhancements ( #22 and #23 ) to take care of the standardization of gameTypes and languages.

File: src/js/pages/Home.js (hunk 5 above)
Comment (wovalle) [reply src/js/pages/Home.js:52]: https://github.com/wistcc/werewolf-moderator/pull/19#issuecomment-292734014

File: src/js/pages/Home.js (hunk 7 above)
Comment (wovalle) [reply src/js/pages/Home.js:51]: https://github.com/wistcc/werewolf-moderator/pull/19#issuecomment-292734014

File: src/js/pages/Cards.js (hunk 4 above)
Comment (wovalle) [reply src/js/pages/Cards.js:52]: https://github.com/wistcc/werewolf-moderator/pull/19#issuecomment-292734014

File: src/js/pages/Cards.js (hunk 6 above)
Comment (wovalle) [reply src/js/pages/Cards.js:51]: https://github.com/wistcc/werewolf-moderator/pull/19#issuecomment-292734014

File: src/js/components/Wizard.js (hunk 3 above)
Comment (wovalle) [reply src/js/components/Wizard.js:5]: https://github.com/wistcc/werewolf-moderator/pull/19#issuecomment-292734014

File: src/js/components/Wizard.js (hunk 1 above)
Comment (wistcc) [reply src/js/components/Wizard.js:32]: This `1` hardcoded?

**Review by wistcc — APPROVED** [review]

File: src/js/components/Wizard.js (hunk 1 above)
Comment (wovalle) [reply src/js/components/Wizard.js:32]: Actually, that's required to be hardcoded (for now). That's the initial amount that a card will have. In order to remove this hardcode number we should have a centralized storage of cards amount (yes, this actually exists in the werewolf brain but the api doesn't return cards amounts).

**Pull request merged**